agent/cache/cache_test.go

```diff
@@ -631,12 +632,15 @@ func TestCacheGet_expire(t *testing.T) {
 	require.Equal(42, result)
 	require.False(meta.Hit)
 
+	time.Sleep(5 * time.Millisecond)
```

Comment:
Sleeps at this granularity will be flaky in CI, unfortunately. I usually only go in increments of 50ms. Since all the tests are set with `t.Parallel` it usually doesn't noticably increase the time for tests since... Amdahl's law.

Reply:
Agreed although I _think_ it's OK here - notice that the only reason for the sleep is not to wait for any baground work to complete but just to ensure that we can measure _at least_ 5ms of age between the fetch taht this thread previously invoked and the following one. So I don't think it will make things flaky it's just sanity checking that the age actually increases by a meaningful amount between two serial calls.

I will add a comment if that isn't clear though as generally for "wait for background" this is unreliably fudge factor.